sources/packages/api/src/sfas-integration/sfas-integration.service.ts

```diff
@@ -3,113 +3,49 @@ import { LoggerService } from "../logger/logger.service";
-   * @param fileName name of the file to be downloaded.
+   * @param remoteFilePath full remote file path with file name.
    * @returns parsed records from the file.
    */
-  async downloadResponseFile(fileName: string): Promise<DownloadResult> {
-    const client = await this.getClient();
-    try {
-      const filePath = this.getFullFilePath(fileName);
-      // Read all the file content and create a buffer.
-      const fileContent = await client.get(filePath);
-      // Convert the file content to an array of text lines and remove possible blank lines.
-      const fileLines = fileContent
-        .toString()
-        .split(/\r\n|\n\r|\n|\r/)
-        .filter((line) => line.length > 0);
-      // Read the first line to check if the header code is the expected one.
-      const header = new SFASHeader(fileLines.shift()); // Read and remove header.
-      if (header.recordType !== RecordTypeCodes.Header) {
-        this.logger.error(
-          `The SFAS file ${fileName} has an invalid transaction code on header: ${header.recordType}`,
-        );
-        // If the header is not the expected one, just ignore the file.
-        return null;
-      }
-
-      // Remove the footer.
-      // Not part of the processing.
-      fileLines.pop();
-
-      // Generate the records.
-      let lineNumber = 2; // 1 is the header already removed.
-      const records = fileLines.map(
-        (line) => new SFASRecordIdentification(line, lineNumber++),
+  async downloadResponseFile(remoteFilePath: string): Promise<DownloadResult> {
+    const fileLines = await this.downloadResponseFileLines(remoteFilePath);
+    // Read the first line to check if the header code is the expected one.
+    const header = new SFASHeader(fileLines.shift()); // Read and remove header.
+    if (header.recordType !== RecordTypeCodes.Header) {
+      this.logger.error(
+        `The SFAS file ${remoteFilePath} has an invalid transaction code on header: ${header.recordType}`,
       );
-
-      return {
-        header,
-        records,
-      };
-    } finally {
-      await SshService.closeQuietly(client);
+      // If the header is not the expected one, throw an error.
+      throw new Error("Invalid file header.");
     }
-  }
 
-  /**
-   * Delete a file from SFTP.
-   * @param fileName name of the file to be deleted.
-   */
-  async deleteFile(fileName: string): Promise<void> {
-    const filePath = this.getFullFilePath(fileName);
-    const client = await this.getClient();
-    try {
-      await client.delete(filePath);
-    } finally {
-      await SshService.closeQuietly(client);
-    }
-  }
+    // Remove the footer.
+    // Not part of the processing.
+    fileLines.pop();
 
-  /**
-   * Generates a new connected SFTP client ready to be used.
-   * @returns client
-   */
-  private async getClient(): Promise<Client> {
-    return this.sshService.createClient(this.ftpConfig);
+    // Generate the records.
+    let lineNumber = 2; // 1 is the header already removed.
```

Comment:
Just to mention, map() also has index param.

Reply:
Changed as per recommendation 😉